src/cg/cst_cg_dump_voice.c

```diff
@@ -88,13 +88,13 @@ static void cst_cg_write_tree_nodes(cst_file fd, const cst_cart_node * nodes)
     while (nodes[num_nodes].val != 0)
         num_nodes++;
 
-    cst_fwrite(fd, &num_nodes, sizeof(int), 1);
+    cst_fwrite(fd, &num_nodes, sizeof(int32_t), 1);
     for (i = 0; i < num_nodes; i++)
     {
         cst_fwrite(fd, &nodes[i].feat, sizeof(char), 1);
         cst_fwrite(fd, &nodes[i].op, sizeof(char), 1);
-        cst_fwrite(fd, &nodes[i].no_node, sizeof(short), 1);
-        cst_fwrite(fd, &nodes[i].val->c.a.type, sizeof(short), 1);
+        cst_fwrite(fd, &nodes[i].no_node, sizeof(int16_t), 1);
```

Comment:
Should the no_node member definition in `cst_cart.h` be changed to int16_t as well since it's accessed as one?